src/Amazon.AspNetCore.Identity.Cognito/CognitoUserManager.cs

```diff
@@ -528,6 +528,49 @@ protected override Task<IdentityResult> UpdateUserAsync(TUser user)
             return _userStore.UpdateAsync(user, CancellationToken);
         }
 
+        /// <summary>
+        ///  Not supported: Returns an IQueryable of users if the store is an IQueryableUserStore
+        /// </summary>
+        public override IQueryable<TUser> Users
+        {
+            get
+            {
+                throw new NotSupportedException("This property is not supported. Use GetUsersAsync() instead.");
+            }
+        }
+
+        /// <summary>
+        /// Queries Cognito and returns the users in the pool. Optional filters can be applied on the users to retrieve based on their attributes.
+        /// Providing an empty attributeFilterName parameter returns all the users in the pool.
+        /// </summary>
+        /// <param name="attributeFilterName"> The attribute name to filter your search on. You can only search for the following standard attributes:
+        ///     username (case-sensitive)
+        ///     email
+        ///     phone_number
+        ///     name
+        ///     given_name
+        ///     family_name
+        ///     preferred_username
+        ///     cognito:user_status (called Status in the Console) (case-insensitive)
+        ///     status (called Enabled in the Console) (case-sensitive)
+        ///     sub
+        ///     Custom attributes are not searchable.
+        ///     For more information, see Searching for Users Using the ListUsers API and Examples
+        ///     of Using the ListUsers API in the Amazon Cognito Developer Guide.</param>
+        /// <param name="attributeFilterType"> The type of filter to apply:
+        ///     For an exact match, use =
+        ///     For a prefix ("starts with") match, use ^=
+        /// </param>
+        /// <param name="attributeFilterValue"> The filter value for the specified attribute.</param>
+        /// <returns>
+        /// The <see cref="Task"/> that represents the asynchronous operation, containing a IEnumerable of CognitoUser.
+        /// </returns>
+        public virtual Task<IEnumerable<CognitoUser>> GetUsersAsync(string filterName = "", string filterType = "", string filterValue = "")
```

Comment:
FilterType should probably be an enum.

FilterName can follow the pattern of "Constant Classes" to have readable constants for each of the currently accepted filter keys, while also allowing users to enter their own if the API expands.